pandas/_libs/window.pyx

```diff
@@ -1599,36 +1346,13 @@ def roll_generic(object obj,
         if not arr.flags.c_contiguous:
             arr = arr.copy('C')
 
-    counts = roll_sum(np.concatenate([np.isfinite(arr).astype(float),
-                                      np.array([0.] * offset)]),
-                      win, minp, index, closed)[offset:]
-
-    start, end, N, win, minp, is_variable = get_window_indexer(arr, win,
-                                                               minp, index,
-                                                               closed,
-                                                               floor=0)
+    counts = roll_sum_fixed(np.concatenate([np.isfinite(arr).astype(float),
```

Comment:
if is is perf-relevant, i expect there is a cnp version of isfinite.

same question before about float vs float64

Reply:
Was getting `TypeError: only size-1 arrays can be converted to Python scalars` from the test suite when trying to use `cnp.math.isfinite` here. I can try to get it working later but I am fairly confident that it won't be a performance bottleneck here.